src/editor/cursor.rs

```diff
@@ -80,6 +80,13 @@ impl Cursor {
             .unwrap_or_else(|| default_colors.foreground.unwrap())
     }
 
+    pub fn blend_is_hide(&self) -> bool {
```

Comment:
maybe rename this to "should_hide"

Also rather than special casing rendering or not, why not have blend apply transparency?

Reply:
![image](https://user-images.githubusercontent.com/3323631/152436734-18503390-b704-4fd2-9f87-b4e00528842f.png)
Specifically the docs indicate that this may be the intended usecase.

Reply:
Actually this is a solution to the background transparency issue we've been having when trying to make the window transparent. I believe this is a great option for that

Reply:
So, just to check, do you mean apply an alpha scaled based on blend between 0 and 100 to `foreground` and `background` in `draw`? Also theres a weird issue where the cursor doesn't go invisible when opening nerdtree (which I think is a simple vsplit), even if `guicursor` is set to something with blend=100, it will update upon switching windows or entering command mode though. I'm guessing it isn't doing a redraw when it should but I don't know much about this codebase yet :p, any pointers?

Reply:
Yes an alpha blend seems like the way to go. The cursor is redrawn on the base canvas with each frame, so it should be fine.

Re redraw. Thats odd, one way to force a draw is to take a reference to https://github.com/neovide/neovide/blob/main/src/redraw_scheduler.rs and then call queue_next_frame. This will force the system to draw what it's got at the next available time. I'd probably try queueing a frame when the renderer receives updated styles and or new cursor